core/src/main/java/jenkins/cli/StopBuildsCommand.java

```diff
@@ -0,0 +1,135 @@
+        final HashSet<String> names = new HashSet<>();
+        names.addAll(jobNames);
+        final StringBuilder resultBuilder = new StringBuilder();
+        for (final String jobName : names) {
+            Item item = jenkins.getItemByFullName(jobName);
+            if (item instanceof Job) {
+                stopJobBuilds((Job) item, jobName, resultBuilder);
+            } else if (item != null) {
+                resultBuilder.append(String.format("Cannot abort runs for %s. Unsupported job type: %s\n", item, item.getClass()));
+            } else {
+                resultBuilder.append(String.format("Job with name %s not found.\n", jobName));
+            }
+        }
+
+        stdout.print(resultBuilder.toString());
+        return 0;
+    }
+
+    private void stopJobBuilds(final Job job,
+                               final String jobName,
+                               final StringBuilder resultBuilder) throws IOException, ServletException {
+        final Run lastBuild = job.getLastBuild();
+        final List<String> stoppedBuildsNames = new ArrayList<>();
+        if (lastBuild != null) {
+            stopBuild(lastBuild, jobName, stoppedBuildsNames);
+            checkAndStopPreviousBuilds(lastBuild, jobName, stoppedBuildsNames);
+        }
+        updateResultOutput(resultBuilder, jobName, stoppedBuildsNames);
+    }
+
+    private void stopBuild(final Run build,
+                           final String jobName,
+                           final List<String> stoppedBuildNames) throws IOException, ServletException {
+        if (build.isBuilding()) {
+            final String buildName = build.getDisplayName();
+            stoppedBuildNames.add(buildName);
+            Executor executor = build.getExecutor();
+            if (executor != null) {
+                executor.doStop();
+                logBuildStopped(jobName, buildName);
+            } else {
+                LOGGER.log(Level.INFO, String.format("Build %s in job %s not stopped", buildName, jobName));
+            }
+        }
+    }
+
+    private void checkAndStopPreviousBuilds(final Run lastBuild,
+                                            final String jobName,
+                                            final List<String> stoppedBuildsNames) throws IOException, ServletException {
+        Run build = lastBuild.getPreviousBuildInProgress();
+        while (build != null) {
+            stopBuild(build, jobName, stoppedBuildsNames);
+            build = build.getPreviousBuildInProgress();
+        }
+    }
+
+    private void updateResultOutput(final StringBuilder result,
+                                    final String jobName,
+                                    final List<String> stoppedBuildNames) {
+        if (stoppedBuildNames.isEmpty()) {
+            result.append(String.format("No builds stopped for job '%s'", jobName));
+        } else {
+            result.append(String.format("Builds stopped for job '%s': ", jobName));
```

Comment:
The use of `StringBuilder` seems unnecessary in this class. You can simply print to `stdout` immediately when doing something. That would make the output more responsive, too, in case the command takes a while to run.